tests/decorators/test_setup_teardown.py

```diff
@@ -255,7 +255,7 @@ def mytask2():
             mytask2()
 
         assert len(dag.task_group.children) == 6
-        assert [x for x in dag.tasks if not x.downstream_list]  # no deps have been set
+        assert any(not x.downstream_list for x in dag.tasks)  # no deps have been set
```

Comment:
Hmm, this test seems wrong actually. The current logic checks there’s at least one task with no deps set, but the comment suggests it should instead test that no deps are set at all.

@ephraimbuddy (author of this) can you confirm which is the original intention here?

Reply:
Not really wrong but can be improved. We should assert that non of the tasks have dependencies probably by checking the total tasks without dependencies:
`assert len([x for x in dag.tasks if not x.downstream_list])==6`